actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -403,25 +406,27 @@ def link_to_unless(condition, name, options = {}, html_options = {}, &block)
       end
 
       # Creates a link tag of the given +name+ using a URL created by the set of
-      # +options+ if +condition+ is true, otherwise only the name is
-      # returned. To specialize the default behavior, you can pass a block that
-      # accepts the name or the full argument list for +link_to_unless+ (see the examples
-      # in +link_to_unless+).
+      # +options+ if +condition+ is true. Otherwise only the name is returned or
+      # the given block is yielded. Therefore you can use the block to specialize
+      # the default behaviour, e.g. change the link target rather (example below).
+      # This behaviour is different from how the block works for +link_to+.
       #
       # ==== Examples
       #   <%= link_to_if(@current_user.nil?, "Login", { controller: "sessions", action: "new" }) %>
       #   # If the user isn't logged in...
       #   # => <a href="/sessions/new/">Login</a>
+      #   # If not...
+      #   # => Login
       #
       #   <%=
       #      link_to_if(@current_user.nil?, "Login", { controller: "sessions", action: "new" }) do
-      #        link_to(@current_user.login, { controller: "accounts", action: "show", id: @current_user })
+      #        link_to("My Profile", { controller: "accounts", action: "show", id: @current_user })
```

Comment:
@senny all right. I will revert it to the previous state. Would you mind when I change it to `@current_user.user_name`? This is not the important part of this PR but I just think it can be more explicit than it is now.